src/common/legacy_config_opts.h

```diff
@@ -1507,6 +1507,8 @@ OPTION(rgw_shard_warning_threshold, OPT_DOUBLE) // pct of safe max
 
 OPTION(rgw_swift_versioning_enabled, OPT_BOOL) // whether swift object versioning feature is enabled
 
+OPTION(rgw_check_http_x_forwarded_for, OPT_BOOL) // allow inspection of HTTP_X_FORWARDED_FOR
```

Comment:
I think this should mean, x-forwarded-for overrides REMOTE_ADDR in bucket policy ip addr restrictions (default true)

Reply:
do we stil need to add legacy config opts unless we're backporting really back?

Reply:
my impression after speaking w @cbodley and @adamemerson was that we should use them for integral types, as there's no expected resolution to perf issues w/the new-style options

Reply:
Do we need a new config? I think Adam has already implemented this logic. 
 In rgw_build_iam_environment, there is a check for REMOTE_ADDR vs HTTP_X_FORWARDED_FOR for the value of rgw remote addr param, based on which the value of aws:SourceIp is set. There is a unit test which tests the same: /ceph/src/test/rgw/test_rgw_iam_policy.cc - TEST_F(IPPolicyTest, IPEnvironment).

Reply:
@adamemerson can correct me if I am wrong.